File: WorldFeed/Client/src/app/core/Core/Models/TwitterEntities/HashTagEntity.ts
import {IHashtagEntity} from "../../../Public/Models/Entities/IHashTagEntity";

/// A hashtag is a keyword prefixed by # and representing a category of tweet
/// This class stores information related with an user mention
export class HashtagEntity implements IHashtagEntity {
  // [JsonProperty("text")]
  public text: string;

  // [JsonProperty("indices")]
  public indices: number[];

  public equals(other: IHashtagEntity): boolean {
    if (other == null || this.text !== other.text) {
      return false;
    }

    if (this.indices == null || other.indices == null) {
      return this.indices === other.indices;
    }

    let arr = ["1", "two"];
    let q = arr.toAddPostfix("fd");

    return this.indices.containsSameObjectsAs(other.indices, true);
  }

  public ToString(): string {
    return `#${this.text}`;
  }
}

File: Sharebook/Server/src/Services/Post/Sharebook.Post.Domain/AggregatesModel/PostAggregate/TwitterEntities/HashTagEntity.cs
namespace Sharebook.Post.Domain.AggregatesModel.PostAggregate.TwitterEntities
{
    using Newtonsoft.Json;

    using Sharebook.Common.Extensions;
    using Sharebook.Post.Domain.AggregatesModel.PostAggregate.Entities;
    using Sharebook.Post.Domain.Common;

    /// <summary>
    /// A hashtag is a keyword prefixed by # and representing a category of tweet
    /// This class stores information related with an user mention
    /// </summary>
    public class HashtagEntity : Entity<long>
    {
        [JsonProperty("text")]
        public string Text { get; set; }

        [JsonProperty("indices")]
        public int[] Indices { get; set; }

        public bool Equals(IHashtagEntity other)
        {
            if (other == null || Text != other.Text)
            {
                return false;
            }

            if (Indices == null || other.Indices == null)
            {
                return Indices == other.Indices;
            }

            return Indices.ContainsSameObjectsAs(other.Indices, true);
        }

        public override string ToString()
        {
            return $"#{Text}";
        }
    }
}
